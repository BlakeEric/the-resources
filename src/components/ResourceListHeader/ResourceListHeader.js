import React, { useContext } from "react"
import { ResourceContext } from '../ResourceContext/ResourceContext'
import { Button } from 'theme-ui'

export default function ResourceListTableHeader({ keys }) {

  const context = useContext(ResourceContext);

  /*
   * Display up or down icon based on key and current order in state
   */
  const displaySortIcon = (keyName) => {
    if (context.orderBy === keyName) {
      if (context.order === "DESC") {
        return '↑'
      }
      return '↓'
    }
    return ''
  }

  /*
   * convert string to uppercase sentence format
   */
  const formatColumnTitle = (string) => {
    let words = string.replace( /([A-Z])/g, " $1" ).split(" ");
    for (let i = 0, x = words.length; i < x; i++) {
      words[i] = words[i][0].toUpperCase() + words[i].substr(1);
    }
    return words.join(" ");
  }

  return (
    <thead >
      <tr>
        {keys.map(keyName => {
          return (
            <th key={keyName} style={{textAlign: "left"}}>
              <Button variant={context.orderBy === keyName ? "secondary" : "muted"} onClick={() => context.setOrder(keyName)}>
                {formatColumnTitle(keyName)} {displaySortIcon(keyName)}
              </Button>
            </th>
          )
        })}
      </tr>
    </thead>
  )
}
