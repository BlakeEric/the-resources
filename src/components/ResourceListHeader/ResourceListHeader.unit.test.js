import React from "react"
import renderer from "react-test-renderer"
import ResourceListHeader from "./ResourceListHeader"

let mockProps = {
  keys: ["name", "skillLevel", "format", "focus"],
  orderBy: "name",
  order: "ASC",
  handleFilterClick: () => {}
}

describe("ResourceListHeader", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ResourceListHeader {...mockProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("Adds appropriate className and icon to current orderBy key", () => {

    mockProps.orderBy = "skillLevel"
    mockProps.order = "DESC"

    const tree = renderer.create(<ResourceListHeader {...mockProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
