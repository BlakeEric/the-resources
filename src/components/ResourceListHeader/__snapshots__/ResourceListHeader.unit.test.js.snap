// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceListHeader Adds appropriate className and icon to current orderBy key 1`] = `
<thead>
  <tr>
    <th
      style={
        Object {
          "textAlign": "left",
        }
      }
    >
      <button
        className="css-vhv7r1"
        onClick={[Function]}
      >
        Name
         
        
      </button>
    </th>
    <th
      style={
        Object {
          "textAlign": "left",
        }
      }
    >
      <button
        className="css-vhv7r1"
        onClick={[Function]}
      >
        Skill Level
         
        ↑
      </button>
    </th>
    <th
      style={
        Object {
          "textAlign": "left",
        }
      }
    >
      <button
        className="css-vhv7r1"
        onClick={[Function]}
      >
        Format
         
        
      </button>
    </th>
    <th
      style={
        Object {
          "textAlign": "left",
        }
      }
    >
      <button
        className="css-vhv7r1"
        onClick={[Function]}
      >
        Focus
         
        
      </button>
    </th>
  </tr>
</thead>
`;

exports[`ResourceListHeader renders correctly 1`] = `
<thead>
  <tr>
    <th
      style={
        Object {
          "textAlign": "left",
        }
      }
    >
      <button
        className="css-vhv7r1"
        onClick={[Function]}
      >
        Name
         
        ↓
      </button>
    </th>
    <th
      style={
        Object {
          "textAlign": "left",
        }
      }
    >
      <button
        className="css-vhv7r1"
        onClick={[Function]}
      >
        Skill Level
         
        
      </button>
    </th>
    <th
      style={
        Object {
          "textAlign": "left",
        }
      }
    >
      <button
        className="css-vhv7r1"
        onClick={[Function]}
      >
        Format
         
        
      </button>
    </th>
    <th
      style={
        Object {
          "textAlign": "left",
        }
      }
    >
      <button
        className="css-vhv7r1"
        onClick={[Function]}
      >
        Focus
         
        
      </button>
    </th>
  </tr>
</thead>
`;
